internal/pubsub/samples/pub/gcp.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2018 The Go Cloud Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+
+	raw "cloud.google.com/go/pubsub/apiv1"
+	"gocloud.dev/gcp"
+	"gocloud.dev/internal/pubsub"
+	"gocloud.dev/internal/pubsub/gcppubsub"
+	"google.golang.org/api/option"
+	"google.golang.org/grpc"
+	grpccreds "google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/oauth"
+)
+
+// openGCPTopic returns the GCP topic based on the topic ID.
+func openGCPTopic(ctx context.Context, topicID string) (*pubsub.Topic, func(), error) {
+	pubClient, cleanup, err := openGCPPubClient(ctx)
+	if err != nil {
+		return nil, nil, err
+	}
+	proj, err := getGCPProjectID()
+	if err != nil {
+		return nil, nil, err
+	}
+	topic := gcppubsub.OpenTopic(ctx, pubClient, proj, topicID, nil)
+	cleanup2 := func() {
+		topic.Shutdown(ctx)
+		cleanup()
+	}
+	return topic, cleanup2, nil
+}
+
+func getGCPProjectID() (gcp.ProjectID, error) {
```

Comment:
FYI, there is a `gcp.DefaultProjectID` function, maybe use that instead of defining your own? It gets them from the credentials instead of the environment which is probably more robust.

Maybe openGCPPubClient / openGCPSubClient could return the `gcp.ProjectID`.

Reply:
Done.

Reply:
This doesn't seem to work well. After making this change, it outputs this error 
```
pub: rpc error: code = NotFound desc = Requested project not found or user does not have access to it (project=). Make sure to specify the unique project identifier and not the Google Cloud Console display name.
```
even after running
```
$ gcloud config set project go-cloud-test-216917
```